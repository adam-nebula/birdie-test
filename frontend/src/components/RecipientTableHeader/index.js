import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";

export default function RecipientTableHeader({tableSchema}) {
    return (
        <TableHead>
            <TableRow>
                {tableSchema.map((field) => (
                    <TableCell align="right" key={"header-" + field.value}>
                        {field.label}
                    </TableCell>
                ))}
            </TableRow>
        </TableHead>
    )
}