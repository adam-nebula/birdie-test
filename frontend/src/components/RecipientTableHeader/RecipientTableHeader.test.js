import React from "react";
import renderer from 'react-test-renderer';
import RecipientTableHeader from ".";

it("recipient table header renders", () => {

    const tableSchema = [{value: "note", label: "Note"}, {value: "timestamp", label: "Timestamp"}];

    const tree = renderer.create(
        <RecipientTableHeader
        tableSchema={tableSchema}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})