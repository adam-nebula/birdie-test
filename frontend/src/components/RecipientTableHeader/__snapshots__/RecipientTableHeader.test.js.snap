// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipient table header renders 1`] = `
<thead
  className="MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Note
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Timestamp
    </th>
  </tr>
</thead>
`;
